### PR TITLE
reimplement move text down and up

### LINKED ISSUE #1056: SPC + x + K raise unexpected behavior after moving to top
## Expected behavior, english is recommended

After the text is moved to the top of the buffer, it should stop there. 
If I press [SPC] + x + K, the moving process should be smooth, while it flashes my editor a lot. 

to be concrete, a video is recorded for reference. [Bug video Recording](https://asciinema.org/a/9RE7qqVM24M4ssEJu6oLGskue)

## Environment Information
- OS: macOS 10.13
- vim version:
- neovim version: 0.2.2

## The reproduce ways from Vim starting (Required!)

Please refer to [Bug video recording](https://asciinema.org/a/9RE7qqVM24M4ssEJu6oLGskue)

## Output of the ':SPDebugInfo!'

```viml
g:spacevim_smartcloseignoreft = ['help', 'tagbar', 'vimfiler', 'SpaceVimRunner', 'SpaceVimQuickFix', 'HelpDescribe']
g:spacevim_default_indent = 2
g:spacevim_custom_plugins = [['terryma/vim-smooth-scroll'], ['chr4/nginx.vim'], ['ryanoasis/vim-devicons']]
g:spacevim_buffer_index_type = 0
g:spacevim_gitcommit_pr_icon = ''
g:spacevim_error_symbol = '✖'
g:spacevim_enable_key_frequency = 0
g:spacevim_max_column = 120
g:spacevim_enable_insert_leader = 1
g:spacevim_vim_help_language = 'en'
g:spacevim_plugin_groups_exclude = []
g:spacevim_dein_installed = 1
g:spacevim_enable_neomake = 1
g:spacevim_relativenumber = 1
g:spacevim_warning_symbol = '⚠'
g:spacevim_realtime_leader_guide = 1
g:spacevim_github_username = ''
g:spacevim_denite_leader = 'F'
g:spacevim_enable_cursorline = 1
g:spacevim_leader_guide_vertical = 0
g:spacevim_statusline_left_sections = ['winnr', 'filename', 'major mode', 'syntax checking', 'minor mode lighters', 'version control info']
g:spacevim_enable_os_fileformat_icon = 1
g:spacevim_leader_guide_position = 'botright'
g:spacevim_version = '0.6.0-dev'
g:spacevim_force_global_config = 0
g:spacevim_enable_javacomplete2_py = 0
g:spacevim_colorscheme = 'gruvbox'
g:spacevim_leader_guide_submode_mappings = {'<C-C>': 'win_close'}
g:spacevim_hiddenfileinfo = 1
g:spacevim_statusline_inactive_separator = 'arrow'
g:spacevim_plugin_manager = 'dein'
g:spacevim_vim_plug_installed = 0
g:spacevim_windows_leader = 's'
g:spacevim_wildignore = '*/tmp/*,*.so,*.swp,*.zip,*.class,tags,*.jpg,*.ttf,*.TTF,*.png,*/target/*,.git,.svn,.hg,.DS_Store,*.svg'
g:spacevim_use_colorscheme = 1
g:spacevim_leader_guide_sort_horizontal = 0
g:spacevim_enable_debug = 0
g:spacevim_enable_vimfiler_gitstatus = 0
g:spacevim_autocomplete_method = 'deoplete'
g:spacevim_plugin_manager_max_processes = 8
g:spacevim_statusline_unicode_symbols = 1
g:spacevim_enable_ycm = 0
g:spacevim_leader_guide_run_map_on_popup = 1
g:spacevim_neobundle_installed = 0
g:spacevim_leader_guide_flatten = 1
g:spacevim_sidebar_width = 30
g:spacevim_enable_googlesuggest = 0
g:spacevim_leader_guide_default_group_name = ''
g:spacevim_terminal_cursor_shape = 2
g:spacevim_snippet_engine = 'neosnippet'
g:spacevim_src_root = 'E:\sources\'
g:spacevim_denite_leader = 'F'
g:spacevim_leader_guide_vertical = 0
g:spacevim_checkinstall = 1
g:spacevim_windows_index_type = 0
g:spacevim_plugin_name = 'gruvbox'
g:spacevim_enable_cursorline = 1
g:spacevim_custom_color_palette = []
g:spacevim_lint_on_the_fly = 0
g:spacevim_debug_level = 1
g:spacevim_filemanager = 'vimfiler'
g:spacevim_language = ''
g:spacevim_enable_vimfiler_gitstatus = 0
```

please post log below, if you want me to reproduce your issue quickly, including your custom config here will be better.
```viml
"let g:one_allow_italics = 1 " I love italic for comments
let g:spacevim_colorscheme_bg = 'dark'
" let g:spacevim_colorscheme = 'one'

"Credit joshdick
"Use 24-bit (true-color) mode in Vim/Neovim when outside tmux.
"If you're using tmux version 2.2 or later, you can remove the outermost $TMUX check and use tmux's 24-bit color support
"(see < http://sunaku.github.io/tmux-24bit-color.html#usage > for more information.)
if (empty($TMUX))
  if (has("nvim"))
  "For Neovim 0.1.3 and 0.1.4 < https://github.com/neovim/neovim/pull/2198 >
  let $NVIM_TUI_ENABLE_TRUE_COLOR=1
  endif
  "For Neovim > 0.1.5 and Vim > patch 7.4.1799 < https://github.com/vim/vim/commit/61be73bb0f965a895bfb064ea3e55476ac175162 >
  "Based on Vim patch 7.4.1770 (`guicolors` option) < https://github.com/vim/vim/commit/8a633e3427b47286869aa4b96f2bfc1fe65b25cd >
  " < https://github.com/neovim/neovim/wiki/Following-HEAD#20160511 >
  if (has("termguicolors"))
    set termguicolors
  endif
endif

set mouse=a

"""""""""""" 
let g:spacevim_enable_vimfiler_welcome = 0
let g:spacevim_enable_tabline_filetype_icon = 1
let g:spacevim_enable_vimfiler_filetypeicon = 1
let g:spacevim_enable_os_fileformat_icon = 1


" If you want to add some custom plugins, use these options:
let g:spacevim_custom_plugins = [
    \ ['terryma/vim-smooth-scroll'],
    \ ['chr4/nginx.vim'], 
    \ ['ryanoasis/vim-devicons'],
    \ ]

" options for vim-devicons
let g:WebDevIconsOS = 'Darwin'

let g:python_host_skip_check=1
let g:python_host_prog = '/usr/bin/python'
let g:python3_host_skip_check=1
let g:python3_host_prog = '/usr/local/bin/python3'
let g:ruby_host_skip_check=1

" disabled plugins
let g:spacevim_disabled_plugins=[
    \ ['tmux-navigator'],
    \ ]

" let g:spacevim_lint_on_save = 1

noremap <silent> <c-u> :call smooth_scroll#up(&scroll, 0, 2)<CR>
noremap <silent> <c-d> :call smooth_scroll#down(&scroll, 0, 2)<CR>
noremap <silent> <c-b> :call smooth_scroll#up(&scroll*2, 0, 4)<CR>
noremap <silent> <c-f> :call smooth_scroll#down(&scroll*2, 0, 4)<CR>



call SpaceVim#layers#load('incsearch')
let g:incsearch#auto_nohlsearch = 0
call SpaceVim#layers#load('autocomplete')
call SpaceVim#layers#load('lang#go')
call SpaceVim#layers#load('lang#python')
call SpaceVim#layers#load('lang#ruby')
call SpaceVim#layers#load('lang#lua')
call SpaceVim#layers#load('tags')

" NeoVim Preview 
set inccommand=nosplit

" Nerdcommenter
let g:NERDCustomDelimiters = {'c': { 'left': '/*', 'rightAlt': '*/' }}


" Default mapping
let g:multi_cursor_next_key='<C-n>'
let g:multi_cursor_prev_key='<C-p>'
let g:multi_cursor_skip_key='<C-x>'
let g:multi_cursor_quit_key='<Esc>'



" Google java formatter 
let g:neoformat_java_google = {
            \ 'exe': 'google-java-format',
            \ 'args': ['-'],
            \ 'stdin': 1,
            \ }
let g:neoformat_enabled_java= ['google']
```

## Screenshots

If you have any screenshots for this issue, please upload here. BTW you can use https://asciinema.org/ for recording video in terminal.

### PR DESCRIPTION
**Pull request by icearith**
reimplement move text down and up, use native move command with force…
close #1056

### CODE REVIEW

**Comment by wsdjeg** [comment]
maybe we need noautocmd， because some plugin listening the TextChanged autocmd.

File: autoload/SpaceVim/layers/edit.vim [hunk 1]
```diff
@@ -211,16 +211,16 @@ function! s:text_transient_state() abort
...
         \ 'exit' : 0,
         \ },
         \ ],
         \ 'right' : [
         \ {
         \ 'key' : 'K',
         \ 'func' : '',
-        \ 'desc' : 'move text up',
-        \ 'cmd' : 'noautocmd exe line(".") == 1 ? "" : "normal! \"_ddkP"',
+        \ 'desc' : 'noautocmd move text up',
```
Comment (wsdjeg) [inline autoload/SpaceVim/layers/edit.vim]: A type? I think the noautocmd should be added to `cmd` instead of `desc`.

Comment (icearith) [reply autoload/SpaceVim/layers/edit.vim]: oh man, what did I do :joy:

**Comment by wsdjeg** [comment]
:+1:

**Pull request merged**